azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -60,37 +61,29 @@ def __init__(self, wire_client):
             self._wire_client = wire_client
 
             # These "basic" properties come from the initial request to WireServer's goalstate API
-            self._timestamp = None
             self._incarnation = None
             self._role_instance_id = None
             self._role_config_name = None
             self._container_id = None
 
-            xml_text, xml_doc = GoalState._fetch_goal_state(self._wire_client)
-
-            self._initialize_basic_properties(xml_doc)
-
-            # The goal state for extensions can come from vmSettings when using FastTrack or from extensionsConfig otherwise, self._fetch_extended_goal_state
-            # populates the '_extensions_goal_state' property.
+            # These "extended" properties come from additional HTTP requests to the URIs included in the basic goal state, or to the HostGAPlugin
             self._extensions_goal_state = None
-            vm_settings = self._fetch_vm_settings()
-
-            # These "extended" properties come from additional HTTP requests to the URIs included in the basic goal state
             self._hosting_env = None
             self._shared_conf = None
             self._certs = None
             self._remote_access = None
 
-            self._fetch_extended_goal_state(xml_text, xml_doc, vm_settings)
+            timestamp = create_timestamp()
+            xml_text, xml_doc, incarnation = GoalState._fetch_goal_state(self._wire_client)
+            self._history = GoalStateHistory(timestamp, incarnation)
+
```

Comment:
the main change here is intantiating the history during __init__; the rest of the changes are just moving the code around to ensure all properties are defined before invoking any method